base/reduce.jl

```diff
@@ -524,6 +476,7 @@ end
     any(itr) -> Bool
 
 Test whether any elements of a boolean collection are `true`.
+Not all items in `itr` will be visited if a `true` value is found.
```

Comment:
The mix of imperative ("test whether") and non-imperative ("returns true") seems a little odd to me but maybe it's fine. How about something like
> Test whether any elements of a boolean collection are `true`, returning `true` as soon as the first `true` value in `itr` is encountered (short-circuiting).

?